sdk/storage/Azure.Storage.Files.DataLake/src/PathClient.cs

```diff
@@ -0,0 +1,1462 @@
+    /// </summary>
+    public class PathClient
+    {
+        /// <summary>
+        /// A <see cref="BlockBlobClient"/> associated with the path;
+        /// </summary>
+        internal readonly BlockBlobClient _blockBlobClient;
+
+        /// <summary>
+        /// BlobClient
+        /// </summary>
+        internal virtual BlockBlobClient BlobClient => _blockBlobClient;
+
+        /// <summary>
+        /// The paths's primary <see cref="Uri"/> endpoint.
+        /// </summary>
+        private readonly Uri _uri;
+
+        /// <summary>
+        /// The paths's blob <see cref="Uri"/> endpoint.
+        /// </summary>
+        private readonly Uri _blobUri;
+
+        /// <summary>
+        /// The path's dfs <see cref="Uri"/> endpoint.
+        /// </summary>
+        private readonly Uri _dfsUri;
+
+        /// <summary>
+        /// DFS Uri
+        /// </summary>
+        internal Uri DfsUri => _dfsUri;
+
+        /// <summary>
+        /// Gets the directory's primary <see cref="Uri"/> endpoint.
+        /// </summary>
+        public virtual Uri Uri => _uri;
+
+        /// <summary>
+        /// The <see cref="HttpPipeline"/> transport pipeline used to send
+        /// every request.
+        /// </summary>
+        private readonly HttpPipeline _pipeline;
+
+        /// <summary>
+        /// Gets the <see cref="HttpPipeline"/> transport pipeline used to send
+        /// every request.
+        /// </summary>
+        internal virtual HttpPipeline Pipeline => _pipeline;
+
+        /// <summary>
+        /// The <see cref="ClientDiagnostics"/> instance used to create diagnostic scopes
+        /// every request.
+        /// </summary>
+        private readonly ClientDiagnostics _clientDiagnostics;
+
+        /// <summary>
+        /// The <see cref="ClientDiagnostics"/> instance used to create diagnostic scopes
+        /// every request.
+        /// </summary>
+        internal virtual ClientDiagnostics ClientDiagnostics => _clientDiagnostics;
+
+        /// <summary>
+        /// The Storage account name corresponding to the directory client.
+        /// </summary>
+        private string _accountName;
+
+        /// <summary>
+        /// Gets the Storage account name corresponding to the directory client.
+        /// </summary>
+        public virtual string AccountName
+        {
+            get
+            {
+                SetNameFieldsIfNull();
+                return _accountName;
+            }
+        }
+
+        /// <summary>
+        /// The file system name corresponding to the directory client.
+        /// </summary>
+        private string _fileSystemName;
+
+        /// <summary>
+        /// Gets the file system name name corresponding to the directory client.
+        /// </summary>
+        public virtual string FileSystemName
+        {
+            get
+            {
+                SetNameFieldsIfNull();
+                return _fileSystemName;
+            }
+        }
```

Comment:
Do you also want the `Path` here?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8419